lib/jxl/decode.cc

```diff
@@ -2041,9 +2045,10 @@ static JxlDecoderStatus HandleBoxes(JxlDecoder* dec) {
       if (have_copy) {
         // TODO(lode): prune the codestream_copy vector if the codestream
         // decoder no longer needs data from previous frames.
-        dec->codestream_copy.insert(dec->codestream_copy.end(), dec->next_in,
+        dec->codestream_copy.insert(dec->codestream_copy.end(),
+                                    dec->next_in + dec->codestream_unconsumed,
                                     dec->next_in + avail_codestream);
-        dec->AdvanceInput(avail_codestream);
+        dec->codestream_unconsumed = avail_codestream;
```

Comment:
Do we assume `dec->codestream_unconsumed == 0`; if yes, lets add `JXL_CHECK` for that.

Reply:
dec->codestream_unconsumed is not necessarily zero here, only if the last call to ProcessInput returned with JXL_DEC_NEED_MORE_INPUT, otherwise we keep copying from where we left off.

Reply:
But `AdvanceInput` was cumulative... and rewriting is not...

Reply:
It is still cumulative, because if we do not consume the input via AdvanceInput, the next time the same input will still be there, possibly with some more bytes at the end (this is required by the API), and then avail_codestream will increase with that same number of bytes. But if we return NEED_MORE_INPUT, then we advance and codestream_unconsumed will be zero.

Reply:
Got it. Thanks for the clarification.